### PR TITLE
Imb 23

### PR DESCRIPTION
**Pull request by muthrajan**
What

Download files from filevault using keycloak access - [IMB-23](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-23) 
Case worker down load  files when they receive email link and the user has already uploaded the files. 

Create and send pdf and submit files added. 

Why

part of IMA MVP

How
Case worker download files when they receive email link and using key cloak access

Testing

Tests passing locally

### CODE REVIEW

**Review by Rhodine-orleans-lindsay — CHANGES_REQUESTED** [review]

File: README.md [hunk 1]
```diff
@@ -134,3 +134,5 @@ $ yarn run test:lint
 ```bash
 $ yarn test:unit
 ```
+
+If you want to run locally make sure you have different ports for Html-pdf-converter and filevault.  In order to download files from the email link received by caseworker, keycloak env variables KEYCLOAK_TOKEN_URL, KEYCLOAK_SECRET, KEYCLOAK_CLIENT_ID, KEYCLOAK_USERNAME and KEYCLOAK_PASSWORD has to be added as provided by keybase.
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply README.md]: README file changed in additional envs section and availability in Keybase removed.

File: apps/ima/translations/src/en/validation.json [hunk 2]
```diff
@@ -1,5 +1,9 @@
 {
-  "who-are-you": {
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/translations/src/en/validation.json:2]: who-are-you validation added

File: apps/ima/views/confirmation.html [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+  <h2 class="govuk-heading-m">{{#t}}pages.confirmation.header2{{/t}}</h2>
+  <p class="govuk-body">{{#t}}pages.confirmation.paragraph2{{/t}}</p>
+
+  <h3 class="govuk-heading-m">{{#t}}pages.confirmation.header3{{/t}}</h2>
+    <p class="govuk-body">{{#t}}pages.confirmation.paragraph4{{/t}}</p>
+    <p class="govuk-body">{{#t}}pages.confirmation.paragraph5{{/t}}</p>
+
+    {{/page-content}}
+    {{/partials-page}}
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/confirmation.html]: New line  added

File: apps/verify/index.js [hunk 4]
```diff
@@ -15,10 +18,14 @@ module.exports = {
     '/verify': {
       fields: ['user-email'],
       behaviours: [SendVerificationEmail],
-      next: '/check-inbox'
+      // next: '/check-inbox' TO BE REINSTATED ONCE RELEVANT PAGES ADDED
+      next: '/ima/evidence-upload' // TO BE REMOVED ONCE RELEVANT PAGES ADDED
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/verify/index.js]: next: '/check-inbox'  is reinstated and next: '/ima/evidence-upload' is removed

File: apps/verify/index.js [hunk 5]
```diff
@@ -5,6 +5,9 @@ const SendVerificationEmail = require('./behaviours/send-verification-email');
 module.exports = {
   name: 'verify',
   baseUrl: '/',
+  pages: {
+    '/email-not-recognised': 'email-not-recognised'
+  },
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/verify/index.js]: pages section removed

**Review by Rhodine-orleans-lindsay — CHANGES_REQUESTED** [review]
just a couple of changes needed (see comments). Could you also squash your commits when ready?

File: apps/ima/behaviours/submit.js [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+const CreateAndSendPDF = require('./create-and-send-pdf');
+
+module.exports = superclass => class extends superclass {
+  successHandler(req, res, next) {
+    const uploadPdfShared = new CreateAndSendPDF({
+      sendReceipt: true,
+      sortSections: true,
+      notifyPersonalisations: {
+        name: req.sessionModel.get('fullName')
+      }
```
Comment (Rhodine-orleans-lindsay) [inline apps/ima/behaviours/submit.js:13]: What is this for?

Comment (muthrajan) [reply apps/ima/behaviours/submit.js]: lines 11-13 removed

File: README.md [hunk 7]
```diff
@@ -115,6 +115,7 @@ The other services used for IMA include
 
 - `PDF_CONVERTER_URL`: If you are running a local PDF converter this is the url and port it is running on. This URL should be in the format `PDF_CONVERTER_URL=http://localhost:<PORT>/convert`
 - `FILEVAULT_URL`: If you are running a filevault locally this is the url and port it is running on. This URL should be in the format `FILEVAULT_URL=http://localhost:<PORT>/file`
+If you want to run locally make sure you have different ports for Html-pdf-converter and filevault.  In order to download files from the email link received by caseworker, keycloak env variables KEYCLOAK_TOKEN_URL, KEYCLOAK_SECRET, KEYCLOAK_CLIENT_ID, KEYCLOAK_USERNAME and KEYCLOAK_PASSWORD has to be added.
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply README.md]: changed to have to be added

File: apps/ima/views/confirmation.html [hunk 8]
```diff
@@ -23,4 +23,5 @@ <h3 class="govuk-heading-m">{{#t}}pages.confirmation.header3{{/t}}</h2>
     <p class="govuk-body">{{#t}}pages.confirmation.paragraph5{{/t}}</p>
 
     {{/page-content}}
-    {{/partials-page}}
\ No newline at end of file
+    {{/partials-page}}
+
```
[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/confirmation.html:27]: New line added

[the remark this answers is not in the document]
Comment (muthrajan) [reply apps/ima/views/confirmation.html]: New line corrected

File: apps/verify/index.js [hunk 9]
```diff
@@ -19,6 +19,9 @@ module.exports = {
     },
     '/check-inbox': {
       behaviours: SendVerificationEmail
+    },
+    '/team-email-invalid': {
+      backLink: 'verify'
```
Comment (Rhodine-orleans-lindsay) [inline apps/verify/index.js:24]: Can you remove this please?

**Review by Rhodine-orleans-lindsay — APPROVED** [review]

**Pull request merged**